src/librustc_mir/transform/check_consts/post_drop_elaboration.rs

```diff
@@ -78,10 +78,10 @@ impl Visitor<'tcx> for CheckLiveDrops<'mir, 'tcx> {
                     return;
                 }
 
-                let needs_drop = if let Some(local) = dropped_place.as_local() {
+                let needs_drop = if !dropped_place.is_indirect() {
```

Comment:
the "not indirect" is a bit confusing. Isn't this basically `dropped_place.is_indirect() || {` and eliminating the `else` arm entirely?